src/main/java/ie/textability/kata/parking/ParkingLot.java

```diff
@@ -0,0 +1,161 @@
+            occupiedSpace.get(ParkingSpotType.REGULAR).add(VehicleType.MOTORCYCLE);
+        } else if(occupiedSpace.get(ParkingSpotType.LARGE).size() < maxCapacity.get(ParkingSpotType.LARGE)) {
+            occupiedSpace.get(ParkingSpotType.LARGE).add(VehicleType.MOTORCYCLE);
+        } else {
+            throw new ParkingLotFullException("FULL");
+        }
+        return this;
+    }
+
+
+    public ParkingLot withCar() {
+        System.out.println("Accommodating "+ VehicleType.CAR +": " + occupiedSpace.get(ParkingSpotType.REGULAR).size());
+        if (occupiedSpace.get(ParkingSpotType.REGULAR).size() < maxCapacity.get(ParkingSpotType.REGULAR)) {
+            occupiedSpace.get(ParkingSpotType.REGULAR).add(VehicleType.CAR);
+        } else if (canAccommodateCarInCompactSpots()) {
+            occupiedSpace.get(ParkingSpotType.COMPACT).add(VehicleType.CAR);
+        } else {
+            throw new RuntimeException("FULL");
+        }
+        return this;
+    }
+
+    private boolean canAccommodateCarInCompactSpots() {
+        if (occupiedSpace.get(ParkingSpotType.COMPACT).size() < maxCapacity.get(ParkingSpotType.COMPACT)) {
+            return true;
+        }
+        return false;
+    }
+
+    public ParkingLot withVan() {
+        // when large spots are full, allow overflow into regular spot allocation
+        System.out.println("Accommodating "+ VehicleType.VAN +": " + occupiedSpace.get(ParkingSpotType.LARGE).size());
+        if (occupiedSpace.get(ParkingSpotType.LARGE).size() < maxCapacity.get(ParkingSpotType.LARGE)) {
+            occupiedSpace.get(ParkingSpotType.LARGE).add(VehicleType.VAN);
+        } else if (canAccommodateVanInRegularSpots()) {
+            occupiedSpace.get(ParkingSpotType.REGULAR).add(VehicleType.VAN);
+            occupiedSpace.get(ParkingSpotType.REGULAR).add(VehicleType.VAN);
+            occupiedSpace.get(ParkingSpotType.REGULAR).add(VehicleType.VAN);
+        } else {
+            throw new RuntimeException("FULL");
+        }
+        return this;
+    }
+
+    private boolean canAccommodateVanInRegularSpots() {
+        if (occupiedSpace.get(ParkingSpotType.REGULAR).size() + 3 <= maxCapacity.get(ParkingSpotType.REGULAR)) {
+            return true;
+        }
+        return false;
+    }
+
+    public int getCompactSpotsCapacity() {
+        return maxCapacity.get(ParkingSpotType.COMPACT);
+    }
+
+    public int getRegularSpotsCapacity() {
+        return maxCapacity.get(ParkingSpotType.REGULAR);
+    }
+
+    public int getLargeSpotsCapacity() {
+        return maxCapacity.get(ParkingSpotType.LARGE);
+    }
+
+    public List<VehicleType> getCompactSpots() {
+        return occupiedSpace.get(ParkingSpotType.COMPACT);
+    }
+
+    public List<VehicleType> getRegularSpots() {
+        return occupiedSpace.get(ParkingSpotType.REGULAR);
+    }
+
+    public List<VehicleType> getLargeSpots() {
+        return occupiedSpace.get(ParkingSpotType.LARGE);
+    }
+
+    /*
+     * Business logic
+     */
```

Comment:
business logic is just a pass-through for CapacityAnalyser. It is probably not needed here, and would reduce coupling.